cmd/fleet/cron.go

```diff
@@ -458,33 +483,74 @@ func cronWorker(
 				break
 			}
 		}
-		if jiraSettings == nil {
-			// currently, Jira is the only job possible, so skip processing jobs if
-			// it is not enabled.
-			level.Debug(logger).Log("msg", "no Jira integration enabled")
-			continue
+
+		// get the enabled zendesk config, if any
+		var zendeskSettings *fleet.ZendeskIntegration
+		for _, intg := range appConfig.Integrations.Zendesk {
+			if intg.EnableSoftwareVulnerabilities {
+				zendeskSettings = intg
+				break
+			}
+		}
+		if jiraSettings != nil && zendeskSettings != nil {
+			level.Error(logger).Log("err", "more than one automation enabled")
+			// TODO: additional error handling?
 		}
 
-		// create the client to make API calls to Jira
-		client, err := externalsvc.NewJiraClient(&externalsvc.JiraOptions{
-			BaseURL:           jiraSettings.URL,
-			BasicAuthUsername: jiraSettings.Username,
-			BasicAuthPassword: jiraSettings.APIToken,
-			ProjectKey:        jiraSettings.ProjectKey,
-		})
-		if err != nil {
-			level.Error(logger).Log("msg", "Error creating Jira client", "err", err)
-			sentry.CaptureException(err)
+		if jiraSettings == nil && zendeskSettings == nil {
+			// skip processing jobs if no integrations are enabled.
+			level.Debug(logger).Log("msg", "no automations enabled")
 			continue
 		}
 
-		// safe to update the Jira worker as it is not used concurrently
-		jira.FleetURL = appConfig.ServerSettings.ServerURL
-		if failerClient != nil && strings.Contains(jira.FleetURL, "fleetdm") {
-			failerClient.JiraClient = client
-			jira.JiraClient = failerClient
-		} else {
-			jira.JiraClient = client
+		if jiraSettings != nil {
+			// create the client to make API calls to Jira
+			client, err := externalsvc.NewJiraClient(&externalsvc.JiraOptions{
+				BaseURL:           jiraSettings.URL,
+				BasicAuthUsername: jiraSettings.Username,
+				BasicAuthPassword: jiraSettings.APIToken,
+				ProjectKey:        jiraSettings.ProjectKey,
+			})
+			if err != nil {
+				level.Error(logger).Log("msg", "Error creating Jira client", "err", err)
+				sentry.CaptureException(err)
+				continue
+			}
+
+			// safe to update the Jira worker as it is not used concurrently
+			jira.FleetURL = appConfig.ServerSettings.ServerURL
+			if jiraFailerClient != nil && strings.Contains(jira.FleetURL, "fleetdm") {
+				jiraFailerClient.JiraClient = client
+				jira.JiraClient = jiraFailerClient
+			} else {
+				jira.JiraClient = client
+			}
```

Comment:
Maybe this should be moved to its own function to make this code shorter? Ditto for zendesk below.